aiida/orm/implementation/general/calculation/job/__init__.py

```diff
@@ -37,6 +36,29 @@ class AbstractJobCalculation(object):
     remotely on a job scheduler.
     """
 
+    _cacheable = True
+
+    @classproperty
+    def _hash_ignored_attributes(cls):
+        # _updatable_attributes are ignored automatically.
+        return super(AbstractJobCalculation, cls)._hash_ignored_attributes + [
+            'queue_name',
+            'max_wallclock_seconds',
+            'retrieve_temporary_list',
```

Comment:
In terms of reducing false positives, wouldn't it be better to also hash `retrieve_temporary_list`? If this list changes, the parser might fail (or return less/more output nodes), so changing the behaviour - and in this case I would say we don't want to cache it. Or more generally, when can the retrieve_temporary_list change without the plugin version change? (Or, if the plugin allows the user to change it, it will have to appear in some input, so it will be hashed). Maybe I'm missing a use case here?

Reply:
Absolutely, I think I somehow misunderstood what ``retrieve_temporary_list`` does.

Reply:
@giovannipizzi After actually running this code on my project, I think ``retrieve_temporary_list`` should be an ``_updatable_attributes``, and therefore ignored. When I launch a calculation without ``retrieve_temporary_list``, it changes to an empty list _after_ the calculation is stored. This means that nodes will not get their original hash when you re-hash them, breaking caching in this case.

Reply:
Since the other "retrieve" attributes are also updatable, I think putting it there is the correct way to solve this:

```
        self._updatable_attributes = (
            'state', 'job_id', 'scheduler_state',
            'scheduler_lastchecktime',
            'last_jobinfo', 'remote_workdir', 'retrieve_list',
            'retrieve_singlefile_list'
        )
```

Reply:
I'll need to look into why this wasn't caught by the tests...

Reply:
Tests didn't catch this because the test which compares actual and stored hashes has a non-zero ``retrieve_temporary_list``. I guess that's ok, we can't test all possible combinations of inputs anyway.

On the other hand, it might be worth investigating why it's possible that an attribute is updatable, without being on the ``updatable_attributes`` list.